File: Midterm/MT3.m
clear;

xdata=load('data.csv'); %load data from file 'data.csv'
T=1;  %sample time
dt=1/128;  %sampling rate 
tvals=(1:(T/dt))*dt;  %Creates a row vector with values from 0 to T with 1/dt divisions

transx=fft(xdata); %FFT of xdata
magx=abs(transx); %take the norm of the FFT of xdata

fxfft=figure;
plot(magx)
xlabel('frequency index (f in units of 1/dt)')
ylabel('X(f)')
saveas(gcf,'MT3_xFFT','epsc')

fx=figure;
plot(tvals,xdata,'.-')
xlabel('t (units of dt)')
ylabel('x(t)')
saveas(gcf,'MT3_xscatter','epsc')

for i=1:length(xdata)-1 %until N-1 due to the size limit in xdata
    yfiltrd(i)=(xdata(i)+xdata(i+1))/2;
end

yfiltrd(length(xdata))=yfiltrd(1); %periodic boundary conditions(BC)

transy=fft(yfiltrd);
magy=abs(transy);

fyfft=figure;
plot(magy)
xlabel('frequency index (f in units of 1/dt)')
ylabel('Y(f)')
saveas(gcf,'MT3_yFFT','epsc')

fy=figure;
plot(tvals,yfiltrd,'.-')
xlabel('t (units of dt)')
ylabel('y(t)')
saveas(gcf,'MT3_yscatter','epsc')

fxy=figure;
plot(tvals,xdata,'.-')
hold on
plot(tvals,yfiltrd,'.-')
xlabel('t (units of dt)')
ylabel('f(t)')
legend({'f(t) = x(t)','f(t) = y(t)'})
saveas(gcf,'MT3_Bothfigs','epsc')
hold off

File: Midterm/MT3.py
import numpy as np
import random
import matplotlib as mp
import matplotlib.pyplot as plt
#from scipy import fft, ifft
from scipy.fftpack import fft,ifft
wall=60
signal=200

T=1
dt=0.001
noiseScale=.01




print("Data 1")  
data1np = np.genfromtxt("data.csv", delimiter=",", skip_header=0)
data1np.shape

xdata=np.array(data1np,dtype=np.float)

npoints=np.size(xdata)

np.size(xdata)
tvals=np.linspace(0,T,npoints)
transx=fft(xdata)
plt.plot(tvals,xdata,'-')
plt.show()
magx=np.abs(transx);
plt.plot(magx,'-')
plt.show()

ydata=np.zeros(np.size(xdata)+1,dtype=np.float)

for iter in range(np.size(xdata)-1):
    ydata[iter]= (xdata[iter]+ xdata[iter+1])/2

ydata[np.size(xdata)]=ydata[0]
tprime=np.linspace(0,T,npoints+1)
plt.plot(tprime,ydata,'-')
plt.show()
transy=fft(ydata)
magy=np.abs(transy);
plt.plot(magy,'-')
plt.show()
